### PR TITLE
[FIRE-59] Added render mode in simulatiion scene

### PR DESCRIPTION
**Pull request by Tluszczyk**

### CODE REVIEW

File: src/main/java/pl/edu/agh/firecell/renderer/BasicRenderer.java [hunk 1]
```diff
@@ -20,6 +20,7 @@ public class BasicRenderer implements Renderer {
     private final IOListener ioListener;
     private final CameraController cameraController;
     private Disposable windowSizeSubscription;
+    private RenderMode mode = RenderMode.Normal;
```
Comment (koszar91) [inline src/main/java/pl/edu/agh/firecell/renderer/BasicRenderer.java:23]:
```suggestion
    private RenderMode renderMode = RenderMode.Normal;
```

File: src/main/java/pl/edu/agh/firecell/renderer/BasicRenderer.java [hunk 2]
```diff
@@ -43,6 +44,11 @@ public void dispose() {
         cameraController.dispose();
     }
 
+    @Override
+    public void setRenderMode(RenderMode mode) {
+        this.mode = mode;
```
Comment (koszar91) [inline src/main/java/pl/edu/agh/firecell/renderer/BasicRenderer.java:49]:
```suggestion
        this.renderMode = mode;
```

File: src/main/java/pl/edu/agh/firecell/renderer/RenderMode.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package pl.edu.agh.firecell.renderer;
+
+public enum RenderMode {
+    Normal, Air_Temperature, Solid_Temperature
```
Comment (koszar91) [inline src/main/java/pl/edu/agh/firecell/renderer/RenderMode.java:4]:
```suggestion
    NORMAL, 
    AIR_TEMPERATURE,
    SOLID_TEMPERATURE
```
That's convetion in Java

**Review by koszar91 — CHANGES_REQUESTED** [review]
Looks good, just naming convention to be changed

**Review by koszar91 — APPROVED** [review]
GG
Edit: checks :p

File: src/main/java/pl/edu/agh/firecell/core/SimulationScene.java [hunk 4]
```diff
@@ -69,6 +70,21 @@ private void renderGUI(double frameTime) {
                 ImGui.text("Framerate: %s".formatted(String.valueOf(Math.round(1 / frameTime))));
                 ImGui.endMenu();
             }
+            if (ImGui.beginMenu("Render mode")) {
+                if (ImGui.menuItem("Normal")) {
+                    renderer.setRenderMode(RenderMode.Normal);
```
Comment (Kamil153) [inline src/main/java/pl/edu/agh/firecell/core/SimulationScene.java:75]: RenderMode values should be uppercase

File: src/main/java/pl/edu/agh/firecell/renderer/BasicRenderer.java [hunk 5]
```diff
@@ -20,6 +20,7 @@ public class BasicRenderer implements Renderer {
     private final IOListener ioListener;
     private final CameraController cameraController;
     private Disposable windowSizeSubscription;
+    private RenderMode renderMode = RenderMode.Normal;
```
Comment (Kamil153) [inline src/main/java/pl/edu/agh/firecell/renderer/BasicRenderer.java:23]: It should be `RenderMode.NORMAL` (uppercase)

**Review by Kamil153 — CHANGES_REQUESTED** [review]

**Pull request merged**